wlms/server.go

```diff
@@ -276,7 +327,47 @@ func (server *Server) InjectGamePingerFactory(gpf GamePingerFactory) {
 	server.gamePingerFactory = gpf
 }
 
-func CreateServerUsing(acceptedConnections chan ReadWriteCloserWithIp, db UserDb, messagesIn chan Message, messagesOut chan Message) *Server {
+// Tells the relay server to start a game with the given name
+// The host position in the game is protected by the given password
+func (server *Server) RelayCreateGame(name string, hostPassword string) bool {
+	// Tell relay to host game
+	var success bool
+	var data rpc_data.NewGameData
+	data.Name = name
+	data.Password = hostPassword
+	err := server.relay.Call("RelayRPC.NewGame", data, &success)
+	if err != nil || success == false {
+		log.Fatal("Unable to create a game on the relay server. This should not happen.")
+		return false
+	}
+	return true
+}
+
+// The relay informs us that the game with the given name has been connected by the host
+func (server *Server) RelayGameConnected(name string) {
+	game := server.HasGame(name)
+	if game == nil {
+		log.Printf("Relay server talks to us about unkown game %s, might already been closed", name)
+		return
+	}
+	game.SetState(*server, CONNECTABLE_BOTH)
+}
+
+// The relay informs us that the game with the given name has been closed
+func (server *Server) RelayGameClosed(name string) {
+	game := server.HasGame(name)
+	if game == nil {
+		log.Print("ERROR: Relay server talks to us about unkown game ", name)
+		return
+	}
+	server.RemoveGame(game)
+}
+
+func (server *Server) GetRelayAddresses() (string, string) {
```

Comment:
Make struct for the adresses that you can return here. It makes the code more readable: `a.ipv4` vs `a[0]`